monoio-http/src/common/body.rs

```diff
@@ -251,6 +254,32 @@ impl Body for HttpBody {
     }
 }
 
+impl futures_core::Stream for HttpBody {
```

Comment:
Here might be problem.

`futures::Stream` works in poll mode, which means it should maintain status inside it self. However, our interfaces are defined in async mode, the status(Future) is returned out and should be kept and wait by the caller.

If we impl `poll_next` as `let fut = next_data(); fut.poll(cx)`, it will drop the future once pending, and the data is lost forever.

One solution is to use `ResueableBoxFuture` like what you have done when modify h2. Another solution is to use monoio's async-style `Stream`(but not work for this case). Another is impl futures::Stream for HttpBody based on its poll interface.

If you decided to use `multer`, I recommend to use the solution 3, which has the best performance.

Reply:
I implemented `ResueableBoxFuture` for now, I couldn't get the Poll interface working, will reach out to you offline